src/decorator.js

```diff
@@ -21,13 +22,30 @@ const createDecorator = (...calculations: Calculation[]): Decorator => (
           if (!isEqual(next, previous)) {
             if (typeof updates === 'function') {
               const results = updates(next, field, values)
-              Object.keys(results).forEach(destField => {
-                form.change(destField, results[destField])
-              })
+
+              if (isPromise(results)) {
+                results.then(resolved => {
```

Comment:
It doesn't seem to handle the case when the promise is rejected?

Reply:
You're right ! But I took example on the `final-form` library where the rejected promises are never handled. I think the user of the library should be responsible to catch rejections and restore the chain : 
```js
{
  field: 'foo', 
  updates: {
    doubleFoo: (fooValue, allValues) => 
      fetchExternalData(fooValue)
        .then(data => {
          throw 'An error occurred'
        })
        .catch(error => {
          // doubleFoo will be set to `null`
          return null
        })
  }
},
```